### DISCUSSION TITLE
“Update” button in ppooll settings

### DISCUSSION THREAD

**Discussion by the-steech** [ppooll-dev/ppooll#65] [Ideas]
Would be cool to have an update button for non-git users to have latest code without going to GitHub.  

Could use "shell" ext to perform "git clone" commands behind the scenes perhaps?

**Comment by the-steech** [comment]
I think this could also come with some sort of version control.  I feel like every change to "main" branch warrants a new version number from ppooll, even if it's like "8.5.2.104" or something.  I can try to build this feature if this is interesting to users

**Comment by chausch** [comment]
that sounds really interesting!!! what do you think @klausfilip ?

**Comment by chausch** [comment]
regarding versioning: klaus gave the newer ppooll versions the same number as the max version it should work on. i added the date number after that for clarification
i.e. 230122 for today …
i know the english / american way would be to start with the month, but ours is sortable, which is a killer feature 😊
we could do something like maxversion.date.iteration
iteration meaning counting up if there's more than one update per day …

**Reply by the-steech** [reply]
I actually prefer the YYMMDD version 😂

**Reply by the-steech** [reply]
@chausch @klausfilip following up on this, i'm looking into writing a "post-push" git hook, so that everytime we push new code to "main", it will update a "version.txt" file with ppooll-version_date_iteration (something like, 8.5.1_23_01_30_0001)

**Reply by klausfilip** [reply]
sounds good

**Comment by chausch** [comment]
btw regarding a script: does git come with every mac & windows machine? because else we'd need a compulsory extra git install. anyway i think this feature would be great.

**Comment by klausfilip** [comment]
hi, yes sure, if there is a way to update ppooll and ppooll_contributions from within ppooll, lets do it.
what is the shell command for this ?
if i understand you correctly we can skip the releases and just go to updates of main ?

**Comment by klausfilip** [comment]
as i understand the git-commands, it needs a local repository that was initially cloned. if users just download ppooll and use it github will promt: fatal: not a git repository (or any of the parent directories): .git

**Comment by the-steech** [comment]
Hmmm I could write a patch that checks for a .git file.  If none exists, I can write commands to connect to the remote repo.  Otherwise, just pull changes :)

**Comment by klausfilip** [comment]
yes please ;)
also to consider is that users who roll their own acts need to put them outside the ppooll package then.
sth like a new ppooll_own_acts folder that is in the searchpath and also needs to be found by ho_st.

**Comment by the-steech** [comment]
a little tricky working with "git" commands in shell... could possibly have it done via "curl" and then "unzip -o" to overwrite
